src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java

```diff
@@ -140,6 +147,83 @@ public String getApiUrl() {
         return apiUrl.toString();
     }
 
+    public String getEmail() throws BayesianException{
+        
+        
+        String url = getOSIOUserUrl();
+        
+        if(url.equals("No-Filter-Found")) {
+        	return "No-Email-Found";
+        }
+        
+        Gson gson;
+        User responseObj;
+        InputStream is = null;
+        BufferedReader br = null;
+
+        HttpGet httpGet = new HttpGet(url);
+        try (CloseableHttpClient client = HttpClients.createDefault();
+                CloseableHttpResponse response = client.execute(httpGet)) {
+    
+        	    HttpEntity entity = response.getEntity();
+                is = entity.getContent();
+                
+                StringBuilder sb = new StringBuilder();
+                String line;
+                br = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+
+                while ((line = br.readLine()) != null) {
+                    sb.append(line);
+                }
+
+                gson = new GsonBuilder().create();
+                 
+                responseObj = gson.fromJson(sb.toString(), User.class);
```

Comment:
We do something similar in this file already: https://github.com/yzainee/fabric8-analytics-jenkins-plugin/blob/d5af23f1f2baa49cebdddbd5c1653388fe53e8bc/src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java#L119-L125

I believe you could reuse the same approach here. `try-with-resources` would also take care of closing the input stream for you.